### ISSUE TITLE
How are we storing the queue in redis?

### ISSUE THREAD

**Issue by kwokwilliam** [kwokwilliam/questionqueue#15]
We are storing it as an object that looks like this right?

```
{"queue": [question object, question object]}
```

**Comment by ljandrea** [comment]
not sure if we actually ever resolved this but this makes sense to me since it would differentiate it from the other session stuff we're storing in redis as well. we should prob make a final decision bout this tho

**Comment by kwokwilliam** [comment]
I think it would make the most sense, that is what I am grabbing it as this, where both key and value are strings

Redis Key: `queue`

Redis Value: `{"queue": [question object, question object]}`

@bryoco Are you modifying the queue at all? Does this work with your code?

**Comment by ljandrea** [comment]
also is that json-encoded string as the redis value? or is it an actual json object?

**Comment by bryoco** [comment]
@kwokwilliam just reviewed my current implementation. This won't be too much work.

Just making sure: I'm updating all three infra containers: redis, mq, and mongo.
Upon receiving a question post from the client, 1) fetches current queue in redis, delete the existing queue structure (_WIP_); 2) publishes to mq (*DONE*); and 3) updates mongo (*DONE*).

Though I believe you can only insert one kv pair at a time to redis, it'd be safer to introduce a lock in case of a spotty network.

**Comment by kwokwilliam** [comment]
Yes that seems correct. Are you deleting the entire queue, or are you just deleting a student inside the queue? I believe you can send the commands in a redis pipeline so that the read and write happen at once, effectively being a lock.

**Comment by bryoco** [comment]
Mmmmmmm I didn't know redis can do that but will look into it!

**Comment by kwokwilliam** [comment]
We used a pipeline as extra credit that one time. I have the code for it if you want to see it.
